### DISCUSSION TITLE
Filagui images

### DISCUSSION THREAD

**Discussion by jnmaloney** [google/filament#4092] [General]
I just built from latest source and I'm still getting a single frame flicker of the texture atlas when I switch to an image.

**Comment by prideout** [comment]
I'm missing some context here, if you filed an Issue then can you link it here?

**Reply by jnmaloney** [reply]
I just noticed it was mentioned in some recent patch notes. I can file an Issue if it's not already.

**Reply by prideout** [reply]
Ah, we fixed an issue that would manifest when an app does `ImGui::Image`.  All the text that was rendered subsequent to the image would be corrupted.  It sounds like you're hitting a different issue, feel free to describe it here or file a bug.

**Reply by jnmaloney** [reply]
Usually when I post something here it's because I'm not sure if there's a bug or some other problem. When I looked closer into this issue it only occurs if the ImGuiHelper is processed inside the Filament loop. I didn't see any flickering when the ImGui commands were set up outside of the Filament loop.

Example attached
[main.zip](https://github.com/google/filament/files/6613386/main.zip)

**Reply by prideout** [reply]
The only stuff between `beginFrame` and `endFrame` should be `Renderer::render`.  The `ImGuiHelper::render` method is meant to be called outside begin / end.

**Reply by dandingol03** [reply]
> Ah, we fixed an issue that would manifest when an app does `ImGui::Image`. All the text that was rendered subsequent to the image would be corrupted. It sounds like you're hitting a different issue, feel free to describe it here or file a bug.

maybe this issue is relevant https://github.com/google/filament/issues/5245? I use filament with `1.9.9` version

**Reply by prideout** [reply]
Yes, that's definitely a bug that we fixed long ago, please use a newer version of FIlament.
